pkg/pod-autoscaler/pkg/pod-resource-updater/controller.go

```diff
@@ -134,12 +134,36 @@ func (c *Controller) runNodeScaleWorker() {
 				return
 			}
 
+			isReady := false
+			for _, c := range pod.Status.Conditions {
+				if c.Type == corev1.PodReady && c.Status == corev1.ConditionTrue {
+					isReady = true
+				}
+			}
+
+			if !isReady {
+				klog.Infof("Pod %s/%s not ready", pod.Namespace, pod.Name)
+				return
+			}
+
 			newPod, err := syncPod(pod, *podScale)
 			if err != nil {
 				klog.Error("Error syncing the pod: ", err)
 				return
 			}
 
+			requireUpdate := false
+			for i := range pod.Spec.Containers {
+				if pod.Spec.Containers[i].Resources.Requests.Cpu().MilliValue() !=
+					newPod.Spec.Containers[i].Resources.Requests.Cpu().MilliValue() {
+					requireUpdate = true
+				}
+			}
+
+			if !requireUpdate {
+				return
```

Comment:
This is a general comment about the whole function. We are performing a for loop on podscales and if any error occurs we call `return`. However this will prevent the resource update of all the remaining pods. Do we really want this? Maybe a `continue` could solve this.